### PR TITLE
ANN: Fix showing error when assigning value to union field

### PR DESCRIPTION
**Pull request by konrad-sz**
changelog: Fix showing error when assigning value to union field outside unsafe block.

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/annotator/RsUnsafeExpressionAnnotator.kt [hunk 1]
```diff
@@ -80,7 +80,8 @@ class RsUnsafeExpressionAnnotator : AnnotatorBase() {
             }
         }
 
-        if (o.fieldLookup != null) {
+        val exprParent = o.parent
+        if (o.fieldLookup != null && (exprParent !is RsBinaryExpr || exprParent.operatorType != AssignmentOp.EQ)) {
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/ide/annotator/RsUnsafeExpressionAnnotator.kt:84]: I suppose this won't work in the case of `a = union0.foo`

**Review by vlad20012 — APPROVED** [review]
bors r+

**Comment by MightyPork** [comment]
This bug is still occuring in plugin version 0.4.200.5421-232. Was the fix not released?

Screenshot_20230828_132140

**Comment by Maksim20023** [comment]
Hi @MightyPork  Thank you for the report.
This fix was included in the Rust plugin 0.4.194 version. Please take a look at change log: https://intellij-rust.github.io/2023/05/08/changelog-194.html
However, please provide a project or code sample with the described behavior, and we will look at what is happening.

**Comment by MightyPork** [comment]
@Maksim20023 This seems to be a distilled version of the bug:

```rust
#![allow(non_camel_case_types)]
#![allow(non_snake_case)]

pub struct BACnet_Application_Data_Value {
    pub type_: BACnet_Application_Data_Value__bindgen_ty_1,
}


pub union BACnet_Application_Data_Value__bindgen_ty_1 {
    pub Special_Event: BACnet_Special_Event,
}

#[derive(Clone, Copy)]
pub struct BACnet_Special_Event {
    pub period: BACnet_Special_Event__bindgen_ty_1,
}


#[derive(Clone, Copy)]
pub union BACnet_Special_Event__bindgen_ty_1 {
    pub calendarReference: BACnet_Object_Id,
}

#[derive(Clone, Copy)]
pub struct BACnet_Object_Id {
    pub type_: u32,
}

fn main() {
    let mut enc: BACnet_Application_Data_Value = unsafe { std::mem::zeroed() };

    // bug here:
    enc.type_.Special_Event.period.calendarReference.type_ = 1;
}
```

It feels like the problem is a nested union.

In the previous screenshot it was also happening for non-nested things, but that cleared now - assuming something bad in my cache.

**Pull request merged**